daisy_workflows/export/disk_resizing_mon.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+METADATA_URL="http://metadata/computeMetadata/v1/instance"
+
+MAX_SIZE=$1
+BUFFER_SIZE=25
+INTERVAL=10
+
+echo "Max disk size ${MAX_SIZE}GB, buffer size ${BUFFER_SIZE}GB, starting monitoring available disk buffer every ${INTERVAL}s..."
+while [ 1 ]
+do
+  sleep ${INTERVAL}
+
+  # Check whether available buffer space is lower than threshold.
+  AVAILABLE_BUFFER=$(df -BG /dev/sdc --output=avail | sed -n 2p)
+  AVAILABLE_BUFFER=${AVAILABLE_BUFFER%?}
+  if [ ${AVAILABLE_BUFFER} -ge ${BUFFER_SIZE} ]; then
+    continue
+  fi
+
+  # Decide the new size, which won't be higher than max size.
+  CURRENT_SIZE=$(lsblk /dev/sdc --output=SIZE | sed -n 2p)
+  CURRENT_SIZE=${CURRENT_SIZE%?}
+  NEXT_SIZE=$(awk "BEGIN {print int(${CURRENT_SIZE} + ${BUFFER_SIZE})}")
+  if [ ${NEXT_SIZE} -gt ${MAX_SIZE} ]; then
+    NEXT_SIZE=${MAX_SIZE}
+  fi
+
+  # Prepare parameters for resizing
+  ZONE=$(curl "${METADATA_URL}/zone" -H "Metadata-Flavor: Google"| cut -d'/' -f4)
+  if [ $? -ne 0 ]; then
+    echo "GCEExport: Failed to get metadata from ${METADATA_URL}/zone, will try again later."
+    continue
+  fi
+  INSTANCE_NAME_PREFIX=$(curl "${METADATA_URL}/attributes/instance-name" -H "Metadata-Flavor: Google")
+  if [ $? -ne 0 ]; then
+    echo "GCEExport: Failed to get metadata from ${METADATA_URL}/attributes/instance-name, will try again later."
+    continue
+  fi
+  HOST_NAME=$(curl "${METADATA_URL}/hostname" -H "Metadata-Flavor: Google"| cut -d'.' -f1)
+  if [ $? -ne 0 ]; then
+    echo "GCEExport: Failed to get metadata from ${METADATA_URL}/hostname, will try again later."
+    continue
+  fi
+  BUFFER_DISK_PREFIX=$(curl "${METADATA_URL}/attributes/buffer-disk" -H "Metadata-Flavor: Google")
+  if [ $? -ne 0 ]; then
+    echo "GCEExport: Failed to get metadata from ${METADATA_URL}/attributes/buffer-disk, will try again later."
+    continue
+  fi
+  BUFFER_DISK=${BUFFER_DISK_PREFIX}-$(echo ${HOST_NAME} | awk -F"${INSTANCE_NAME_PREFIX}-" '{ print $2}')
+
+  echo "GCEExport: Resizing buffer disk from ${CURRENT_SIZE}GB to ${NEXT_SIZE}GB..."
+  if ! out=$(gcloud compute disks resize ${BUFFER_DISK} --size=${NEXT_SIZE}GB --quiet --zone "${ZONE}" 2>&1); then
```

Comment:
you only need to capture the output to a variable if you will use it conditionally or in another command..you echo it in both failure and success cases, so just remove the subshell wrapping

Reply:
or dont echo it on success..

Reply:
In error path, the output needs to be embedded in a "ExportFailed:" msg, which is required for daisy to capture. It works as a "thrown exception" somehow. So I do need this variable.

Reply:
the requirement is to print 'ExportFailed' at beginning of line on failure. that's orthogonal to the capture strategy here.. you can do:
```
if ! gcloud compute ....; then
  echo ExportFailed: failed to export
fi
```

Reply:
I understand the point. However, I need to avoid newline after echo. I did a bit research, it seems that different implementation of 'echo' has inconsistent support on this. I want to get rid of the dependency on that. Does that sound good?

Reply:
offline synced, will change to double bracket as style guide suggested.

Reply:
there are two `echo`, a binary and a bash built-in. they differ in handling of escapes, but not in handling of newlines. if you're concerned, specify `/bin/echo`, but i think it's not related to the substance of my comment. if you always want output, simply remove `out=$()` wrapping

Reply:
In this script, "out" must be in the same line with "ExportFailed", otherwise daisy can't show it to user on console. I do need the $out var to achieve this.